src/sage/rings/species.py

```diff
@@ -0,0 +1,2255 @@
+
+            S = SymmetricGroup(sum(self.grade())).young_subgroup(self.grade())
+            # conjugate self and other to match S
+            G, G_dompart = self.group_and_partition()
+            g = list(chain.from_iterable(G_dompart))
+            conj_self = PermutationGroupElement(g).inverse()
+            G = libgap.ConjugateGroup(G, conj_self)
+            H, H_dompart = other.group_and_partition()
+            h = list(chain.from_iterable(H_dompart))
+            conj_other = PermutationGroupElement(h).inverse()
+            H = libgap.ConjugateGroup(H, conj_other)
+            return GAP_FAIL != libgap.ContainedConjugates(S, G, H, True)
+
+        def __lt__(self, other):
+            r"""
+            Return if this element is less than ``other``.
+
+            ``self`` is less than or equal to ``other`` if it is
+            conjugate to a subgroup of ``other`` in the parent group.
+
+            EXAMPLES::
+
+                sage: from sage.rings.species import MolecularSpecies
+                sage: M = MolecularSpecies("X")
+                sage: M(SymmetricGroup(4)) < M(CyclicPermutationGroup(4))
+                True
+            """
+            if (not isinstance(other, MolecularSpecies.Element)
+                or len(self.grade()) != len(other.grade())):
+                return False
+            return self != other and self <= other
+
+        def __gt__(self, other):
+            r"""
+            Return if this element is greater than ``other``.
+
+            ``self`` is less than or equal to ``other`` if it is
+            conjugate to a subgroup of ``other`` in the parent group.
+
+            EXAMPLES::
+
+                sage: from sage.rings.species import MolecularSpecies
+                sage: M = MolecularSpecies("X")
+                sage: M(SymmetricGroup(4)) > M(CyclicPermutationGroup(4))
+                False
+            """
+            if (not isinstance(other, MolecularSpecies.Element)
+                or len(self.grade()) != len(other.grade())):
+                return False
+            return other < self
+
+        def __ge__(self, other):
+            r"""
+            Return if this element is greater than or equal to ``other``.
+
+            ``self`` is less than or equal to ``other`` if it is
+            conjugate to a subgroup of ``other`` in the parent group.
+
+            EXAMPLES::
+
+                sage: from sage.rings.species import MolecularSpecies
+                sage: M = MolecularSpecies("X")
+                sage: M(SymmetricGroup(4)) >= M(CyclicPermutationGroup(4))
+                False
+            """
+            if (not isinstance(other, MolecularSpecies.Element)
+                or len(self.grade()) != len(other.grade())):
+                return False
+            return other <= self
+
+        def group_and_partition(self):
+            """
+            Return the (transitive) permutation group corresponding to ``self``.
```

Comment:
Wouldn't it be better to call this method `[associated_]permutation_group()`? In either case, it should be in the documentation that it also returns the (set) partitioned domain. Or perhaps that should be an optional parameter?